events/events.go

```diff
@@ -246,13 +247,14 @@ func (d *DisconnectSender) PublishDisconnect(startTime time.Time, err error) {
 	}
 
 	status := StatusFailure
-	if err != nil {
+	if err == nil {
 		status = StatusSuccess
 	}
 
 	d.eventTemplate.EventStatus = status
 	d.eventTemplate.Duration = time.Since(startTime)
 	d.eventTemplate.Error = err
+	d.eventTemplate.IsRefresh = true
```

Comment:
I guess this should be easily unit  tested. Can you please provide some tests to cover that scenario?